arviz/data/io_pymc3.py

```diff
@@ -68,29 +77,36 @@ def arbitrary_element(dct: Dict[Any, np.ndarray]) -> np.ndarray:
             # if you have a posterior_predictive built with keep_dims,
             # you'll lose here, but there's nothing I can do about that.
             self.nchains = 1
-            aelem = (
-                arbitrary_element(prior)
-                if posterior_predictive is None
-                else arbitrary_element(posterior_predictive)
-            )
+            get_from = None
+            if predictions is not None:
+                get_from = predictions
+            elif prior is not None:
+                get_from = prior
+            elif posterior_predictive is not None:
+                get_from = posterior_predictive
```

Comment:
I would keep the priority order with posterior predictive before prior: `predictions`, `posterior_predictive`, `prior`. 

This is because the default in `pm.sample_posterior_predictive` is `samples=None` which will end up with ndraw samples, however, for `pm.sample_prior_predictive` the default is `samples=500` which will generally not be equal to ndraws.

Reply:
I have a tentative version of the above and will push it momentarily. Note that I have augmented the `orig_idata` by hand instead of using `concat`.  LMK if you think `concat` would be better.